set_env.bat

```diff
@@ -0,0 +1,29 @@
+@echo off
+REM
+REM Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+REM or more contributor license agreements. Licensed under the Elastic License
+REM 2.0 and the following additional limitation. Functionality enabled by the
+REM files subject to the Elastic License 2.0 may only be used in production when
+REM invoked by an Elasticsearch process with a license key installed that permits
+REM use of machine learning features. You may not use this file except in
+REM compliance with the Elastic License 2.0 and the foregoing additional
+REM limitation.
+REM
+
+REM Set up a build environment, to ensure repeatable builds
+
+REM Initialize the Visual Studio command prompt environment variables
+cmd /c "C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+
+REM Set %CPP_SRC_HOME% to be an absolute path to this script's location, as
+REM different builds will come from different repositories and go to different
+REM staging areas
+SET MY_DIR=%~dp0
+SET CPP_SRC_HOME=%~dp0
+
+REM Logical filesystem root
+if NOT "%LOCAL_DRIVE%"== "" (SET ROOT=%LOCAL_DRIVE%) else (SET ROOT=C:)

```

Comment:
Instead of hardcoding `C:` this should be able to use `%CD:~0,2%`.  Then it really is the local drive, whatever that happens to be.

This is what I was getting at in the comments about this on other PRs - using native Windows functionality we should not have to specify the current drive letter because Windows knows what it is.  It's a side effect of using Cygwin/MinGW that the paths incorporate the drive letters in a way that's not natural to Windows.